### ISSUE TITLE
RedundantImport does not work with module imports

### ISSUE THREAD

**Issue by stoyanK7** [checkstyle/checkstyle#18171]
Part of issue
- https://github.com/checkstyle/checkstyle/issues/18127
---
I have read check documentation: https://checkstyle.org/checks/imports/redundantimport.html#RedundantImport
I am using the latest checkstyle generated from the `master` branch via `mvn clean package -Passembly`
I have executed the CLI and showed it below, as CLI describes the problem better than 1,000 words

```bash
$ javac -version
javac 25.0.1

$ javac Test.java
# no output as expected

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="RedundantImport"/>
  </module>
</module>

$ cat Test.java
import module java.base;
import module java.base;  // I expect violation here, but there is none.

class Test {
}

$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
$ java $RUN_LOCALE -jar checkstyle-12.2.0-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```
---
I expect to see an error that tells me that I have a redundant import `java.base`. Something like:
> [ERROR] /tmp/Test.java:2:1: Duplicate import to line 1 - java.base [RedundantImport]

**Comment by stoyanK7** [comment]
If approved, I am on it.

**Comment by romani** [comment]
Do you plan compare import of modules independently from regular imports? 
Do you target simply searching duplicates of imports ?

I thought this Check is about to find imports that can be removed due no such class referenced in class

**Comment by stoyanK7** [comment]
> Do you plan compare import of modules independently from regular imports?

Yes. This is already the case. `RedundantImport` treats regular imports (`import ...`) and static imports (`import static ...`) independently:
https://github.com/checkstyle/checkstyle/blob/48980d94bf918c2e57be0e465c403af667fd80f3/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/RedundantImportCheck.java#L107-L133

>  Do you target simply searching duplicates of imports ?

Yes, _only duplicate imports_. Similar to how static imports are handled in the above code snippet.

> I thought this Check is about to find imports that can be removed due no such class referenced in class

I believe you are mixing the check in this issue ([`RedundantImport`](https://checkstyle.org/checks/imports/redundantimport.html#RedundantImport)) with [`UnusedImports`](https://checkstyle.org/checks/imports/unusedimports.html#UnusedImports). `UnusedImports` is the one that finds such unreferenced imports:
> # UnusedImports
> ## Description
>  Checks for unused import statements. An import statement is considered unused if:
>    - It is not referenced in the file.

**Comment by stoyanK7** [comment]
@romani 

> Do you plan compare import of modules independently from regular imports?

Yes. This is already the case. `RedundantImport` treats regular imports (`import ...`) and static imports (`import static ...`) independently:
https://github.com/checkstyle/checkstyle/blob/48980d94bf918c2e57be0e465c403af667fd80f3/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/RedundantImportCheck.java#L107-L133

>  Do you target simply searching duplicates of imports ?

Yes, _only duplicate imports_. Similar to how static imports are handled in the above code snippet.

> I thought this Check is about to find imports that can be removed due no such class referenced in class

I believe you are mixing the check in this issue ([`RedundantImport`](https://checkstyle.org/checks/imports/redundantimport.html#RedundantImport)) with [`UnusedImports`](https://checkstyle.org/checks/imports/unusedimports.html#UnusedImports). `UnusedImports` is the one that finds such unreferenced imports:
> # UnusedImports
> ## Description
>  Checks for unused import statements. An import statement is considered unused if:
>    - It is not referenced in the file.

**Comment by romani** [comment]
Yes, you are right. Issue is approved.

**Comment by stoyanK7** [comment]
On it.

**Comment by romani** [comment]
fix is merged
